src/ServiceFabricPersistence/SynchronizedStorage/SynchronizedStorageFeature.cs

```diff
@@ -1,17 +1,17 @@
 ﻿namespace NServiceBus.Persistence.ServiceFabric
 {
     using Features;
+    using Microsoft.Extensions.DependencyInjection;
 
     class SynchronizedStorageFeature : Feature
     {
         protected override void Setup(FeatureConfigurationContext context)
         {
             var stateManager = context.Settings.StateManager();
             var transactionTimeout = context.Settings.TransactionTimeout();
-            
-            var container = context.Container;
-            container.ConfigureComponent(b => new SynchronizedStorage(stateManager, transactionTimeout), DependencyLifecycle.SingleInstance);
-            container.ConfigureComponent(b => new SynchronizedStorageAdapter(), DependencyLifecycle.SingleInstance);
+
+            context.Services.AddSingleton<ISynchronizedStorage>(b => new SynchronizedStorage(stateManager, transactionTimeout));
+            context.Services.AddSingleton<ISynchronizedStorageAdapter>(b => new SynchronizedStorageAdapter());
```

Comment:
at first glance it looks like we don't need the factories here. But I'm also good to keep them to keep changes minimal.

Reply:
you're right